addons/mail/static/src/chatter/web/chatter_patch.js

```diff
@@ -261,6 +305,13 @@ patch(Chatter.prototype, {
         this.load(thread, ["suggestedRecipients"]);
     },
 
+    onUpdateProps(nextProps) {
+        if (this.isThreadShifted) {
+            this.changeThread(nextProps.threadModel, nextProps.threadId, nextProps.webRecord);
+        }
+        super.onUpdateProps(nextProps);
```

Comment:
Can't you keep changeThread (a basic version with insert) in the basic code to avoid this extra method, as well as having to define isThreadShifted?

Reply:
`changeThread()` accepts a parameter as `webRecord` and we don't have `webRecord` in portal. This is why I didn't keep it there. Although I can remove the `isThreadShifted()` and just repeat the condition in the web part.